networkx/algorithms/node_classification.py

```diff
@@ -170,8 +169,9 @@ def local_and_global_consistency(G, alpha=0.99, max_iter=30, label_name="label")
     # Build propagation matrix
     degrees = X.sum(axis=0)
     degrees[degrees == 0] = 1  # Avoid division by 0
-    # TODO: csr_array
-    D2 = np.sqrt(sp.sparse.csr_array(sp.sparse.diags((1.0 / degrees), offsets=0)))
+    D2 = np.sqrt(
+        sp.sparse.dia_array((1.0 / degrees, 0), shape=(n_samples, n_samples)).tocsr()
+    )
```

Comment:
Not related to this PR, but it's probably more efficient to apply the sqrt to `degrees` rather than the full array as is done in `algebraicconnectivity`.